ecosystem/typescript/sdk/examples/typescript/gas_fee_payer.ts

```diff
@@ -0,0 +1,228 @@
+} from "aptos";
+import { NODE_URL, FAUCET_URL } from "./common";
+import { assert } from "console";
+
+const client = new AptosClient(NODE_URL);
+const tokenClient = new TokenClient(client);
+const faucetClient = new FaucetClient(NODE_URL, FAUCET_URL);
+
+async function waitForEnter() {
+  return new Promise<void>((resolve, reject) => {
+    const rl = require("readline").createInterface({
+      input: process.stdin,
+      output: process.stdout,
+    });
+
+    rl.question("Please press the Enter key to proceed ...\n", () => {
+      rl.close();
+      resolve();
+    });
+  });
+}
+
+async function ensureTxnSuccess(txnHashPromise: Promise<string>) {
+  const txnHash = await txnHashPromise;
+  const txn = await client.waitForTransactionWithResult(txnHash);
+  assert((txn as any)?.success);
+}
+
+const getBalance = async (account: AptosAccount) => {
+  const resources = await client.getAccountResources(account.address().hex());
+  const aptosCoin = "0x1::coin::CoinStore<0x1::aptos_coin::AptosCoin>";
+  let accountResource = resources.find((r) => r.type === aptosCoin);
+  return BigInt((accountResource!.data as any).coin.value);
+};
+
+async function createTokenWithFeePayer(feePayer: AptosAccount, account: AptosAccount, collectionName: string, name: string, description: string, supply: number, uri: string, max: BCS.AnyNumber, royalty_payee_address?: MaybeHexString, royalty_points_denominator?: number, royalty_points_numerator?: number, property_keys?: Array<string>, property_values?: Array<string>, property_types?: Array<string>, extraArgs?: OptionalTransactionArgs): Promise<string> {
+  // Build the raw transaction.
+  const builder = new TransactionBuilderRemoteABI(client, { sender: account.address().hex(), ...extraArgs });
+  const rawTxn = await builder.build(
+    "0x3::token::create_token_script",
+    [],
+    [
+      collectionName,
+      name,
+      description,
+      supply,
+      max,
+      uri,
+      royalty_payee_address,
+      royalty_points_denominator,
+      royalty_points_numerator,
+      [false, false, false, false, false],
+      property_keys,
+      getPropertyValueRaw(property_values, property_types),
+      property_types,
+    ],
+  );
+  // Build the fee payer transaction.
+  const feePayerTxn = new TxnBuilderTypes.FeePayerRawTransaction(
+    rawTxn,
+    [],
+    TxnBuilderTypes.AccountAddress.fromHex(feePayer.address().hex()),
+  );
+  // `account` signs the fee payer raw transaction.
+  const accountSignature = new TxnBuilderTypes.Ed25519Signature(
+    account.signBuffer(TransactionBuilder.getSigningMessage(feePayerTxn)).toUint8Array(),
+  );
+  // Construct the account authenticator.
+  const accountAuthenticator = new TxnBuilderTypes.AccountAuthenticatorEd25519(
+    new TxnBuilderTypes.Ed25519PublicKey(account.signingKey.publicKey),
+    accountSignature,
+  );
+  // `feePayer` signs the fee payer raw transaction.
+  const feePayerSignature = new TxnBuilderTypes.Ed25519Signature(
```

Comment:
I wonder if this flow works out of the box with wallets. cc @0xmaayan @kent-white do wallets/wallet connector currently support signing multi-agent transactions?

Reply:
I just talked to @hardsetting about it, if I understand correctly it should be supported, but defer to @hardsetting

Reply:
from the wallet adapter side, I believe it supports only a payload input so it assumes wallet knows how to handle it to generate and submit a multi transaction - might need to do some adapter changes?

Reply:
I don't have the context. What do you mean by the wallet supporting the gas fee payment flow? What is a use case scenario?

Reply:
The wallet + wallet adapter need to support these specific lines
```
const accountSignature = new TxnBuilderTypes.Ed25519Signature(
    account.signBuffer(TransactionBuilder.getSigningMessage(feePayerTxn)).toUint8Array(),
  );
```

@0xmaayan I'm a bit rusty on what the wallet adapter interface looks like. Is there a way to sign something like this? You can also give us some code pointers as well and we can go dig into it

Reply:
the adapter is just a layer to pass data/info to the wallet. It does have functions like `signMessage` / `signTransaction` but at the end it depends on the wallet implementation
https://github.com/aptos-labs/aptos-wallet-adapter/blob/main/packages/wallet-adapter-core/src/WalletCore.ts#L329-L372

Reply:
Sounds good. @junkil-park I think the best next step here is to test this out in the browser and see if wallet connection works. Perhaps you can do some simple tweakings around Mayaan's e2e dapp guide https://aptos.dev/tutorials/build-e2e-dapp/create-a-smart-contract to load your code in the browser and test out the wallet connection part.